diff -Naur LC3plus-1.6.1.old/src/floating_point/makefile LC3plus-1.6.1/src/floating_point/makefile
--- LC3plus-1.6.1.old/src/floating_point/makefile	2026-03-17 10:26:18.961995110 +0100
+++ LC3plus-1.6.1/src/floating_point/makefile	2026-03-17 10:26:36.255646811 +0100
@@ -25,6 +25,9 @@
 # Binary Name
 NAME_LC3   ?= LC3plus
 # Shared Library Name
+MAJOR      = 1
+MINOR      = 5
+MICRO      = 1
 LIB_LC3    ?= libLC3plus.so
 
 # Default tool settings
@@ -141,10 +144,12 @@
 
 $(LIB_LC3): $(LIBOBJS)
 	@echo 'Linking' $@
-	$(QUIET) $(LINK) --shared $(OBJS) -o $@ $(LDFLAGS)
+	$(QUIET) $(LINK) --shared -Wl,-soname=$(@F).$(MAJOR) -o $@.$(MAJOR).$(MINOR).$(MICRO) $(OBJS) $(LDFLAGS)
+	ln -sf $(@F).$(MAJOR).$(MINOR).$(MICRO) $@.$(MAJOR)
+	ln -sf $(@F).$(MAJOR) $@
 
 clean:
-	$(QUIET) rm -rf $(NAME_LC3) $(LIB_LC3) $(BUILD) ccConvert
+	$(QUIET) rm -rf $(NAME_LC3) $(LIB_LC3).* $(BUILD) ccConvert
 
 $(BUILD)/%.o : %.c $(BUILD)/cc_flags
 	@echo 'Compiling' $<
